fleetspeak/src/server/https/client_certificate_test.go

```diff
@@ -140,35 +171,98 @@ func TestFrontendMode_HEADER_TLS(t *testing.T) {
 	clientCertHeader := "ssl-client-cert"
 	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		// test the valid frontend mode combination of receiving the client cert in the header
-		cert, err := GetClientCert(req, clientCertHeader, cpb.FrontendMode_HEADER_TLS)
+		cert, err := GetClientCert(req, clientCertHeader, cpb.FrontendMode_HEADER_TLS, "")
+		if err != nil {
+			t.Fatal(err)
+		}
+		// make sure we received the client cert in the header
+		if cert == nil {
+			t.Error("Expected client certificate but received none")
+		}
+		// test the invalid frontend mode combination with frontend mode HEADER_TLS_CHECKSUM
+		cert, err = GetClientCert(req, clientCertHeader, cpb.FrontendMode_HEADER_TLS_CHECKSUM, "")
+		if err == nil {
+			t.Error("Expected error for invalid frontend mode combination but received none")
+		}
+		// test the invalid frontend mode combination with frontend mode MTLS
+		cert, err = GetClientCert(req, clientCertHeader, cpb.FrontendMode_MTLS, "")
+		if err == nil {
+			t.Error("Expected error for invalid frontend mode combination but received none")
+		}
+		fmt.Fprintln(w, "Testing Frontend Mode: HEADER_MTLS")
+	}))
+	ts.TLS = &tls.Config{
+		ClientAuth: tls.RequireAnyClientCert,
+        }
+	ts.StartTLS()
+	defer ts.Close()
+
+	_, client, bc, _ := makeTestClient(t)
+
+	clientCert := url.PathEscape(string(bc))
+	req, err := http.NewRequest(http.MethodGet, ts.URL, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set(clientCertHeader, clientCert)
+
+	res, err := client.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	body, err := io.ReadAll(res.Body)
+	res.Body.Close()
```

Comment:
nit: found the position of the `Close()` a bit weird xD
We usually `defer` right after "opening", whenever we want to; or in the case of tests, we can use `t.Cleanup`

Reply:
Moved the closer to "opening" as `defer`.